setup/native-script.rb

```diff
@@ -23,32 +27,73 @@
+      break if show_all_option ? is_answer_yn || $answer == "all" : is_answer_yn
+    end
+
+    if $answer == "n"
+      puts "You've chosen not to install " + program_name + " some features of NativeScript may not work correctly if you haven't already installed it"
+    end
+  end
+
+  puts message
+  execute(script, program_name + " not installed", run_as_root)
+end
+
+# Actually installing all other dependencies
+install("Homebrew",	"Installing Homebrew...", 'ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"</dev/null', false, false)
 
 if !(`brew --version`.include? "git revision")
   puts "Homebrew is not installed or not configured properly. Download it from http://brew.sh/, install, set it up and run this script again."
   exit
 end
 
-puts "Installing CocoaPods... This might take some time, please, be patient. (You might need to provide your password.)"
-system('sudo gem install cocoapods')
-
-puts "Installing Homebrew Cask... (You might need to provide your password.)"
-system('brew install caskroom/cask/brew-cask')
+install("Java SE Development Kit", "Installing the Java SE Development Kit... This might take some time, please, be patient.(You will be prompted for your password)", 'brew cask install java', false, false)
+execute('echo "export JAVA_HOME=$(/usr/libexec/java_home)" >> ~/.profile', "Unable to set JAVA_HOME")
 
-puts "Installing the Java SE Development Kit... This might take some time, please, be patient. (You might need to provide your password.)"
-system('brew cask install java')
-system('echo "export JAVA_HOME=$(/usr/libexec/java_home)" >> ~/.profile')
+install("Android SDK", "Installing Android SDK", 'brew install android-sdk')
+execute('echo "export ANDROID_HOME=/usr/local/opt/android-sdk" >> ~/.profile', "Unable to set ANDROID_HOME")
 
-puts "Installing Android SDK"
-system('brew install android-sdk')
-system('echo "export ANDROID_HOME=/usr/local/opt/android-sdk" >> ~/.profile')
+# the -p flag is set in order to ensure zero status code even if the directory exists
+execute("mkdir -p ~/.cocoapods", "There was a problem in creating ~/.cocoapods directory")
+install("CocoaPods", "Installing CocoaPods... This might take some time, please, be patient.", 'gem install cocoapods -V', true)
 
 puts "Configuring your system for Android development... This might take some time, please, be patient."
-system "echo yes | /usr/local/opt/android-sdk/tools/android update sdk --filter tools,platform-tools,android-23,build-tools-23.0.2,extra-android-m2repository --all --no-ui"
+# Note that multiple license acceptances may be required, hence the multiple y answers
+# the android tool will introduce a --accept-license option in subsequent releases
+execute("(for i in {1..5}; do echo y; sleep 4; done) | /usr/local/opt/android-sdk/tools/android update sdk --filter tools,platform-tools,android-23,build-tools-23.0.2,extra-android-m2repository --all --no-ui",
```

Comment:
I haven't encountered any issues regarding the installation of all the dependencies (apart from accepting licenses in this weird fashion)